packages/daemon/src/pet-store.js

```diff
@@ -85,12 +85,14 @@ export const makePetStoreMaker = (filePowers, locator) => {
       assertValidName(petName);
       assertValidFormulaIdentifier(formulaIdentifier, petName);
 
+      // TODO: Return early if the formula identifier is the same.
       if (petNames.has(petName)) {
         // Perform cleanup on the overwritten pet name.
         const formulaPetNames = formulaIdentifiers.get(petName);
         if (formulaPetNames !== undefined) {
           formulaPetNames.delete(petName);
         }
+        // TODO: Should this only happen if something is actually deleted?
```

Comment:
I noticed these couple oddities in the pet store's `write()`. I'll create an issue if we agree that there's something to do.

Reply:
@kriskowal thoughts on these?

Reply:
It would greatly simplify client code if they are assured that they only see `delete` messages for equal but opposite prior `add`.